dataprofiler/profilers/float_column_profile.py

```diff
@@ -105,23 +105,39 @@ def _get_float_precision(cls, df_series):
         :return: string representing its precision print format
         :rtype: int
         """
-        integer_decimal_loc = -1
-        float_precision = None
+        if df_series.empty:
+            return 0
+
+        # set it to first value length, at very least will be smaller since min.
+        precision = len(df_series.iloc[0])
         for value in df_series:
-            decimal_loc = value.rfind('.')
 
-            # integers will not have a '.'
-            if decimal_loc == integer_decimal_loc:
-                continue
+            # remove all spaces around value
+            value = value.strip()
+
+            # if scientific notation, remove e and everything after
+            e_ind = value.find('e')
+            if e_ind > -1:
+                value = value[:e_ind]
+
+            # remove negative of positive characters
+            if value[0] in ['+', '-']:
+                value = value[1:]
```

Comment:
Couldn't this technically be behind the number as well, i.e. `1+e3`?

Reply:
Is that a valid notation? That looks more like an equation `1 + 1000`

Reply:
Sorry, misplaced the sign on the wrong side of the `e`, but yes this is: `1e+3`

https://encyclopedia2.thefreedictionary.com/E+notation

> A type of scientific notation in which the phrase “times 10 to the power of” is replaced by the letter E; for example, 3.1 × 10^7 is written 3.1E+7 and 5.1 × 10-9 is written 5.1E-9.

So I was slightly off, but very close and is a common notation. All through school I used the notation in the style. `5.1e-9` notation.